mtags/src/main/scala-3/scala/meta/internal/pc/ScalaPresentationCompiler.scala

```diff
@@ -449,7 +450,10 @@ case class ScalaPresentationCompiler(
       item.setDocumentation(hoverContent(None, None, documentation))
     }
 
-    item.setDeprecated(completion.symbols.forall(_.isDeprecated))
+    if (completion.symbols.forall(_.isDeprecated)) {
+      item.setTags((item.getTags.asScala.toSet ++ Set(CompletionItemTag.Deprecated)).toList.asJava)
```

Comment:
I think `tags` can be null here and that is failing in the tests. Weirdly enough it causes the whole suite to hang, which should not happen, especially that each tests is max 10 minutes :thinking: 

Since we create the item in this method we can just do:
```scala
  item.setTags(List(CompletionItemTag.Deprecated).asJava) 
```

Reply:
@tgodzik would you be opposed to this?

```
item.setTags((Option(item.getTags).map(_.asScala.toSet).getOrElse(Set.empty) ++ Set(CompletionItemTag.Deprecated)).toList.asJava)
```

I think `  item.setTags(List(CompletionItemTag.Deprecated).asJava) ` is fine for today, but since I'm assuming lsp4j is changing this API to allow for more types of tags, I can foresee a future bug where we set the tags in one place, then overwrite them in another.

Reply:
@tgodzik I've updated to check for `null`. Let me know if you strongly prefer

```
 item.setTags(List(CompletionItemTag.Deprecated).asJava) 
 ```
 
 If so, I'll change it to that.

Reply:
Item is really created in the same method just above, so I don't think we benefit much by doing so. If anyone wants to move it at some point they will have to take a look at what's going on anyway.

I think it's really impossible to future-proof it and this might actually never change.

Reply:
I'll update it to the overwriting version you suggested.